app/models/batch_form.rb

```diff
@@ -28,6 +28,20 @@ def self.human_attribute_name(*args)
     Batch.human_attribute_name(*args)
   end
 
+  if Rails::VERSION::MAJOR >= 6
+    include ActiveModel::Attributes
+    attribute :date_produced, :date
+  else
+    def date_produced=(value)
+      value = value.presence
+      if value.is_a?(Time) || value.nil?
+        @date_produced = value
+      else
+        @date_produced = value.to_time rescue nil
+      end
+    end
+  end
```

Comment:
What about moving the setter to the `DateProduced` concern instead of the duplication?

Reply:
I briefly considered that, but wasn't sure The setter is only necessary for a lacking attributes API. It wouldn't be necessary when the field is backed by a database column with datetime type. So one could consider it as being specific to the implementing type, not the concern.
However, the form types themselves don't even include `DateProduced`.
And I think by now we can probably get rid of that concern entirely. The only function that it has left is about display logic which is trivial and really doesn't belong into the model.
I'm just waiting for the next PR to resolve this =)